src/index/TextIndexReadWrite.h

```diff
@@ -273,7 +271,47 @@ class FrequencyEncode {
 
   // View must be an input range with value type `T`.
   template <typename View>
-  explicit FrequencyEncode(View&& view);
+  requires(!std::same_as<FrequencyEncode, std::remove_cvref_t<View>>)
+  explicit FrequencyEncode(View&& inputView) {
```

Comment:
Please move this to the .cpp-File again.

1. C++20 requires-clauses must be repeatet in the definition, so 
in the .cpp-file you woul also have to write
```
template <typename View> requires(!std:...) FrequencyEncode(View&&...) {...
```

2 We are currently backporting QLever to C++ 17, so we use macros that use concepts only in C++17 mode. Please use the `CPP_template` and `CPP_template_def` macros, see [here](https://github.com/ad-freiburg/qlever/wiki/Porting-QLever-to--Cpp17) for details.
So in the header, you write
`CPP_template(typename View)(requires !std::same_as...) exlicit...`
And in the .cpp file
`CPP_template_def(typename View> (requires ...) FrequencyEncode...`

(same for the `GapEncode` class.